pkg/templates/sample/report/ReportTesterProcessor.gen.go

```diff
@@ -0,0 +1,20 @@
+package istio_mixer_adapter_sample_report
+
+import "istio.io/mixer/pkg/adapter/config"
+
+const TemplateName = "istio.mixer.adapter.sample.report.Sample"
+
+// Instance represent the runtime structure that will be passed to the ReportSample method in the handlers.
+type Instance struct {
+	Name       string
+	Value      interface{}
+	Dimensions map[string]interface{}
+}
+
+// SampleProcessor represent the Go interface that handlers must implement if it wants to process the template
+// named `istio.mixer.adapter.sample.report.Sample`
+type SampleProcessor interface {
+	config.Handler
+	ConfigureSample(map[string]*Type /*Constructor:instance_name to Type mapping. Note type name will not be passed at all*/) error
+	ReportSample([]*Instance /*The type is inferred from the Instance.name and the mapping of instance to types passed during the config time*/) error
```

Comment:
So, the content on the Type and Instance (we might make instance a proto eventually) might contain pointers under it... so even a plain value will not help, since adapters can tweak the content of the ref object
(.. we will have to do a deep copy.)

I suppose we are not going to make a new copy of instance (from a handlers.instances) for a given request call
. If we make a new instance for each handler, then we should not care if they decide to manipulate the content of the instance object since it is their copy.. 

@ZackButcher @geeknoid : Given all that, do you guys think we make this a pass by value ?